org-code-javabuilder/lib/src/main/java/org/code/javabuilder/AWSContentManager.java

```diff
@@ -0,0 +1,183 @@
+package org.code.javabuilder;
+
+import com.amazonaws.AbortedException;
+import com.amazonaws.HttpMethod;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import org.code.protocol.ContentManager;
+import org.code.protocol.InternalErrorKey;
+import org.code.protocol.JavabuilderException;
+import org.code.protocol.LoggerUtils;
+
+public class AWSContentManager implements ContentManager, ProjectFileLoader {
+  // Temporary limit on writes per session until we can more fully limit usage.
+  private static final int WRITES_PER_SESSION = 2;
+  // Limit on the amount of S3 file uploads per session to avoid spamming
+  private static final int UPLOADS_PER_SESSION = 20;
```

Comment:
Can you add a bit more detail to these two comments - either here or by the equivalent exception in `UserInitiatedExceptionKey` (or both) - I was having a hard time differentiating why one or the other might be used until I realized that one gets used for prompter and the other gets used by theater. I also noticed that the exceptions don't get interpreted on the Java Lab side. Mind adding either a fallback message or a Java Lab message for those exceptions?

Reply:
(The exceptions are TOO_MANY_UPLOADS and TOO_MANY_WRITES, below)

Reply:
Add comments and a fallback message for TOO_MANY_UPLOADS. For TOO_MANY_WRITES, I think it makes sense to display the standard error message for now ("Oops. We hit an error on our side...") because a student should only hit that error under very strange circumstances, such as Stage.play() is somehow called more than once per session without throwing and exception, or if they're calling AWS APIs directly from their code somehow (which shouldn't be possible with our custom class loader). In either case i'd imagine we'd want to investigate in more detail on our end.